packages/cozy-konnector-libs/src/libs/error.js

```diff
@@ -1,19 +1,55 @@
 const log = require('cozy-logger').namespace('Error Interception')
 
-// This will catch exception which would be uncaught by the connector script itself
-process.on('uncaughtException', err => {
+const handleUncaughtException = err => {
   log('critical', err.message, 'uncaught exception')
   process.exit(1)
-})
-process.on('unhandledRejection', err => {
+}
+
+const handleUnhandledRejection = err => {
   log('critical', err.message, 'unhandled exception')
   process.exit(1)
-})
-process.on('SIGTERM', () => {
+}
+
+const handleSigterm = () => {
   log('critical', 'The konnector got a SIGTERM')
   process.exit(128 + 15)
-})
-process.on('SIGINT', () => {
+}
+
+const handleSigint = () => {
   log('critical', 'The konnector got a SIGINT')
   process.exit(128 + 2)
-})
+}
+
+let attached = false
+
+/**
+ * Attach event handlers to catch uncaught exceptions/rejections and signals.
+ * Log them as critical and exit the process accordingly.
+ * If the cleanup function has not been called, calling again the function
+ * is a no-op.
+ *
+ * @param  {Process} prcs - Process object, default to current process
+ * @return {Function} When called, removes the signal handlers
+ */
+const attachProcessEventHandlers = (prcs = process) => {
```

Comment:
Why do you pass a prcs parameters when global process is used in event handlers anyway ?

Reply:
For tests.